packages/webapp/src/components/Forms/GeneralRevenue/index.jsx

```diff
@@ -12,93 +12,151 @@
  *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
-import React from 'react';
-import { useForm } from 'react-hook-form';
+import React, { useEffect, useState } from 'react';
+import { useForm, Controller } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
 import Form from '../../Form';
 import Button from '../../Form/Button';
 import Input, { getInputErrors } from '../../Form/Input';
 import InputAutoSize from '../../Form/InputAutoSize';
 import PageTitle from '../../PageTitle/v2';
+import ReactSelect from '../../Form/ReactSelect';
+import { IconLink } from '../../Typography';
+import { ReactComponent as TrashIcon } from '../../../assets/images/document/trash.svg';
+import DeleteBox from '../../Task/TaskReadOnly/DeleteBox';
 import { getLocalDateInYYYYDDMM } from '../../../util/date';
 import { hookFormMaxCharsValidation } from '../../Form/hookformValidationUtils';
-
-const SALE_DATE = 'sale_date';
-const SALE_CUSTOMER = 'customer_name';
-const VALUE = `value`;
-const NOTE = `note`;
+import { SALE_DATE, SALE_CUSTOMER, VALUE, NOTE, REVENUE_TYPE_ID } from './constants';
+//import useHookFormPersist from '../../../containers/hooks/useHookFormPersist';
 
 const GeneralRevenue = ({
   onSubmit,
   title,
-  dateLabel,
-  customerLabel,
   currency,
   sale,
   useHookFormPersist,
   persistedFormData,
-  onClickDelete,
+  view,
+  handleGoBack,
+  onClick,
+  revenueTypeOptions,
+  onTypeChange,
+  buttonText,
+  customFormChildrenDefaultValues,
+  useCustomFormChildren = () => {},
+  revenueType,
+  revenueTypes,
+  onRetire,
 }) => {
   const { t } = useTranslation();
+  const [isDeleting, setIsDeleting] = useState(false);
 
-  const {
-    register,
-    handleSubmit,
-    formState: { errors, isValid },
-    getValues,
-  } = useForm({
+  const reactHookFormFunctions = useForm({
     mode: 'onChange',
     defaultValues: {
       [SALE_DATE]:
         (sale?.[SALE_DATE] && getLocalDateInYYYYDDMM(new Date(sale[SALE_DATE]))) ||
         persistedFormData?.[SALE_DATE] ||
         getLocalDateInYYYYDDMM(),
       [SALE_CUSTOMER]: sale?.[SALE_CUSTOMER] || persistedFormData?.[SALE_CUSTOMER] || '',
-      [VALUE]: !isNaN(sale?.[VALUE]) ? sale[VALUE] : null,
-      [NOTE]: sale?.[NOTE] || '',
+      [REVENUE_TYPE_ID]:
+        revenueTypeOptions?.find((option) => option.value === sale?.revenue_type_id) ||
+        revenueTypeOptions?.find(
+          (option) => option.value === persistedFormData?.[REVENUE_TYPE_ID],
+        ) ||
+        null,
+      [VALUE]: !isNaN(sale?.[VALUE])
+        ? sale[VALUE]
+        : !isNaN(persistedFormData?.[VALUE])
+        ? persistedFormData[VALUE]
+        : null,
```

Comment:
This double ternary is a bit hard to read, maybe we could consider making this a separate function with if/else?

Reply:
yeah I think I complicated this one 😂

Reply:
fixed I think.